src/main/java/com/gaggle/contact/controllers/ContactController.java

```diff
@@ -44,6 +46,20 @@
                 .orElseThrow(() -> new ContactNotFoundException(name));
     }
 
+    @GetMapping("/query")
+    public List<Contact> getByEmail(@RequestParam final String email) {
+        if (Objects.isNull(email) || !email.contains("@")) {
+            throw new RuntimeException("email is no good, try again silly");
+        }
+
+        Optional<List<Contact>> contacts = repository.findByEmailContaining(email);
+        if (contacts.get().isEmpty()) {
```

Comment:
Based on customer requests the isEmpty check can be removed and just return the empty list

Reply:
Use size() instead; get() can be process heavy when checking lists.